### PR TITLE
Add --binaryen-test-out to allow setting the test output dir.

### PR DESCRIPTION
**Pull request by kripken**
This allows fuzzing in parallel invocations of `fuzz_opt.py`.

### CODE REVIEW

**Comment by kripken** [comment]
Not the test suite, but the fuzzer, `fuzz_opt.py` (which uses functionality in the test suite). Running multiple invocations lets one fuzz using multiple cores.

File: scripts/test/shared.py [hunk 1]
```diff
@@ -57,6 +57,11 @@ def parse_args(args):
         '--binaryen-root', dest='binaryen_root', default='',
         help=('Specifies a path to the root of the Binaryen repository tree.'
               ' Default: the directory where this file check.py resides.'))
+    parser.add_argument(
+        '--binaryen-test-out', dest='binaryen_test_out', default='',
```
Comment (sbc100) [inline scripts/test/shared.py]: I think simply calling this `--out-dir` would be more clear and concise.    The binaryen-test part is kind of redundant

File: scripts/test/shared.py [hunk 2]
```diff
@@ -124,7 +129,8 @@ def warn(text):
 
 options.binaryen_test = os.path.join(options.binaryen_root, 'test')
 
-test_out = os.path.join(options.binaryen_root, 'out', 'test')
+test_out = (options.binaryen_test_out or
+            os.path.join(options.binaryen_root, 'out', 'test'))
```
Comment (sbc100) [inline scripts/test/shared.py]: How about:

```
if not options.out_dir:
   options.out_dir = os.path.join(options.binaryen_root, 'out', 'test')
```
```

**Comment by kripken** [comment]
Ok, updated according to those suggestions, PTAL @sbc100

**Pull request merged**